### ISSUE TITLE
DropGestureRecognier.Drop Event is not Triggered

### ISSUE THREAD

**Issue by BrandonStudio** [dotnet/maui#20153]
### Description

Two phenomina:
1.  Handler indicated in XAML is marked gray (meaning it is not referenced)
2.  Even when manually added that handler in construct method, the handler is not triggered.

Additional info:
After `InitializeComponents`, handler indicated in XAML is actually added. (But just not triggered)

### Steps to Reproduce

XAML:
```xml
<Label Text="Here">
    <Label.GestureRecognizers>
        <DropGestureRecognizer x:Name="recognizer" Drop="DropGestureRecognizer_Drop" />
    </Label.GestureRecognizers>
</Label>
```

C#:
```csharp
private async void DropGestureRecognizer_Drop(object? sender, DropEventArgs e)
{
    Debug.WriteLine("Triggered");
}
```

### Version with bug

8.0.3

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows, I was *not* able test on other platforms

### Did you find any workaround?

Not yet

**Comment by jaosnz-rep** [comment]
Can repro this issue at Windows platform on the latest 17.10 preview2 (8.0.14 &8.0.7)

**Comment by rmarinho** [comment]
Duplicated of #6080

**Comment by nappa0326** [comment]
Any progress on this issue?

**Comment by jonathan-m-phillips** [comment]
> They are not completely same. #6080 does not say that Drop event is not triggered.

Right. And that was merged, but I have the same exact issue mentioned in this, which was closed. My events are not triggering. I cannot log them or even hit them with break points. I copied line for line from the documentation.

https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/gestures/drag-and-drop?view=net-maui-8.0&tabs=macios#enable-drop
